src/pudl/transform/ferc714.py

```diff
@@ -0,0 +1,749 @@
+    df = (
+        df.rename(columns=_hours_to_ints)
+        .melt(
+            id_vars=["report_year", "utility_id_ferc714",
+                     "report_date", "utc_offset_code"],
+            var_name="hour",
+            value_name="demand_mwh")
+    )
+    _log_dupes(df, ["utility_id_ferc714", "report_date", "hour"])
+
+    df = (
+        df.pipe(_standardize_offset_codes, tz_fixes=TZ_FIXES)
+        # Discard records lacking *both* UTC offset code and non-zero demand
+        # In practice, this should be *all* of the XXX records, but we're being
+        # conservative, in case something changes / goes wrong. We want to
+        # notice if for some reason later we find XXX records that *do* have
+        # real demand associated with them.
+        .query("utc_offset_code!='XXX' | demand_mwh!=0.0")
+        # Almost all 25th hours are unusable (0.0 or daily totals),
+        # and they shouldn't really exist at all based on FERC instructions.
+        .query("hour!=25")
+        # Switch to using 0-23 hour notation, and combine hour with date:
+        .assign(
+            local_time=lambda x:
+                x.report_date + pd.to_timedelta(x.hour - 1, unit="h")
+        )
+    )
+
+    # Set a few year-specific offset codes:
+    df.loc[(df.report_year == 2006) & (
+        df.utility_id_ferc714 == 139), "utc_offset_code"] = "PST"
+    df.loc[(df.report_year == 2015) & (
+        df.utility_id_ferc714 == 235), "utc_offset_code"] = "MST"
+    df.loc[(df.report_year == 2011) & (
+        df.utility_id_ferc714 == 289), "utc_offset_code"] = "CST"
+    df.loc[(df.report_year == 2011) & (
+        df.utility_id_ferc714 == 292), "utc_offset_code"] = "CST"
+
+    # Flip the sign on two sections of demand which were reported as negative:
+    mask1 = (df.report_year.isin([2006, 2007, 2008, 2009])) & (
+        df.utility_id_ferc714 == 156)
+    df.loc[mask1, "demand_mwh"] = -1.0 * df.loc[mask1, "demand_mwh"]
+    mask2 = (df.report_year.isin([2006, 2007, 2008, 2009, 2010])) & (
+        df.utility_id_ferc714 == 289)
+    df.loc[mask2, "demand_mwh"] = -1.0 * df.loc[mask2, "demand_mwh"]
+
+    df = (
+        df.pipe(_to_utc_and_tz, offset_codes=OFFSET_CODES, tz_codes=TZ_CODES)
+        .pipe(_complete_demand_timeseries)
+        .loc[:, [
+            "report_year",
+            "utility_id_ferc714",
+            "utc_datetime",
+            "timezone",
+            "demand_mwh"
+        ]]
+        .sort_values(["utility_id_ferc714", "utc_datetime"])
+    )
+    tf_dfs["pa_demand_hourly_ferc714"] = df
+    return tf_dfs
+
+
+def electricity_planning_areas(pudl_settings):
+    """Electric Planning Area geometries from HIFLD."""
+    gdb_path = pathlib.Path(
+        pudl_settings["data_dir"],
+        "local/hifld/electric_planning_areas.gdb"
+    )
+
+    gdf = (
+        geopandas.read_file(gdb_path)
+        .assign(
+            SOURCEDATE=lambda x: pd.to_datetime(x.SOURCEDATE),
```

Comment:
are these all caps column names? if so... why?

Reply:
These are the original column names coming out of the HIFLD geometry file, which we don't really have a transform module for, and which I don't think we will end up using, except to validate the geometries we generate for 2018.  It just didn't seem worth renaming everything on something disposable.